wire-ios-sync-engine/Source/SessionManager/UpdateEventMigrator.swift

```
@@ -210,6 +220,62 @@ private extension UpdateEvent {
         )
     }
 
+    private static func coversationMemberLeaveEvent(
+        from event: ZMUpdateEvent,
+        localDomain: String
+    ) -> ConversationMemberLeaveEvent? {
+        let decoder = EventPayloadDecoder()
+        guard
+            let payload = try? decoder.decode(
+                Payload.ConversationEvent<Payload.UpdateConversationMemberLeave>.self,
+                from: event.payload
+            ),
+            let conversationID = payload.conversationID(localDomain: localDomain),
+            let senderID = payload.senderID(localDomain: localDomain),
+            let timestamp = payload.timestamp,
+            let leaveReason = payload.data.reason
+        else {
+            return nil
```

question: is there a way to know about the error at some level?

---

We could catch a decoding error throw it up, or we could throw a `missingData` error for the rest?

---

done 698e84594df52360b3650adda0c2d03795aa99d3